src/charm.py

```diff
@@ -1082,16 +1056,23 @@ def render_pgb_config(self, reload_pgbouncer=False, restart=False) -> None:
                 )
         logger.info("pushed new pgbouncer.ini config files to pgbouncer container")
 
-        if reload_pgbouncer:
-            self.reload_pgbouncer(restart)
+        logger.info(f"{'restarting' if restart else 'reloading'} pgbouncer application")
 
-    def render_auth_file(self, auth_file: str, reload_pgbouncer=False):
-        """Renders the given auth_file to the correct location."""
-        self.push_file(AUTH_FILE_PATH, auth_file, 0o400)
-        logger.info("pushed new auth file to pgbouncer container")
+        pebble_services = pgb_container.get_services()
+        try:
+            self.push_file(auth_file, userlist, 0o400)
+            for service in self._services:
+                if service["name"] not in pebble_services:
+                    # pebble_ready event hasn't fired so pgbouncer has not been added to pebble config
+                    raise PebbleConnectionError
+                if restart or pebble_services[service["name"]].current != ServiceStatus.ACTIVE:
+                    pgb_container.restart(service["name"])
+                else:
+                    pgb_container.send_signal(SIGHUP, service["name"])
+        finally:
+            self.delete_file(auth_file)
```

Comment:
Delete the file. We should most probably add additional cleanup in update status.